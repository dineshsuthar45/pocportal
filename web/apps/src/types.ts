import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
