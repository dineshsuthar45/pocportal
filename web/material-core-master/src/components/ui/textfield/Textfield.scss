.textfield-root {
	.MuiIconButton-root {
		margin-right: 0px;
	}
}
