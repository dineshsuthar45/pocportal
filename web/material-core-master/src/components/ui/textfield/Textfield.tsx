import { IconButton, InputAdornment, useTheme } from "@mui/material";
import TextField from "@mui/material/TextField";
import classNames from "classnames";
import { useState } from "react";
import { v4 as uuidv4 } from "uuid";
import { MIconEye, MIconEyeClosed, MIconEyeClosedO, MIconEyeO } from "../icons";
import "./Textfield.scss";
import { TextfieldProps } from "./Textfield.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/text-field/
 */
export const MTextfield = ({
  autoFocus = false,
  color = "primary",
  disabled = false,
  error = false,
  fullWidth = false,
  id = uuidv4(),
  margin = "none",
  required = false,
  size = "medium",
  type = "text",
  variant = "outlined",
  startAdornment,
  endAdornment,
  ...props
}: TextfieldProps) => {
  const theme = useTheme();
  const [showPassword, setShowPassword] = useState<boolean>(false);

  const handleClickShowPassword = () => {
    setShowPassword(!showPassword);
  };

  const handleMouseDownPassword = (
    event: React.MouseEvent<HTMLButtonElement>
  ) => {
    event.preventDefault();
  };

  const openEyeIcon =
    theme.palette.mode === "light" ? <MIconEye /> : <MIconEyeO />;
  const closedEyeIcon =
    theme.palette.mode === "light" ? <MIconEyeClosed /> : <MIconEyeClosedO />;

  const endAdornmentComponent =
    endAdornment ??
    (type === "password" && (
      <InputAdornment position="end">
        <IconButton
          aria-label="toggle password visibility"
          onClick={handleClickShowPassword}
          onMouseDown={handleMouseDownPassword}
          edge="end"
        >
          {showPassword ? openEyeIcon : closedEyeIcon}
        </IconButton>
      </InputAdornment>
    ));

  return (
    <TextField
      className={classNames("textfield-root", theme.palette.mode)}
      autoFocus={autoFocus}
      color={color}
      disabled={disabled}
      error={error}
      fullWidth={fullWidth}
      id={id}
      margin={margin}
      required={required}
      size={size}
      type={type === "password" && showPassword ? "text" : type}
      variant={variant}
      InputProps={{
        startAdornment: startAdornment,
        endAdornment: endAdornmentComponent,
      }}
      {...props}
    />
  );
};

MTextfield.defaultProps = {
  autoFocus: false,
  color: "primary",
  disabled: false,
  error: false,
  fullWidth: false,
  id: uuidv4(),
  margin: "none",
  required: false,
  size: "medium",
  type: "text",
  variant: "outlined",
};
