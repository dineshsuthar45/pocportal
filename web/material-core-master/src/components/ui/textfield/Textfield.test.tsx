import { render } from "@testing-library/react";
import React from "react";
import { MTextfield } from "./Textfield";
import { TextfieldProps } from "./Textfield.types";

describe("Textfield", () => {
  const renderComponent = () => render(<MTextfield {...props} />);
  let props: TextfieldProps;

  beforeEach(() => {
    props = {
      id: "txt1",
      label: "Textfield label",
      onChange: () => {},
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
