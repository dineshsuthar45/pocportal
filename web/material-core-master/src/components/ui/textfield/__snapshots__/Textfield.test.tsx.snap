// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield should render correctly 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfield-root light css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for="txt1"
      id="txt1-label"
    >
      Textfield label
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="txt1"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-yjsfm1"
        >
          <span>
            Textfield label
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
